depsync/depsync.go

```diff
@@ -202,9 +282,19 @@ SYNC:
 			if err := w.publisher.Publish(context.TODO(), SyncTopic, data); err != nil {
 				return err
 			}
+			go func() {
+				err := w.processAutoAssign(resp.Devices)
+				if err != nil {
+					level.Info(w.logger).Log("err", err, "auto-assign")
+				}
+			}()
 		}
 		if !resp.MoreToFollow {
-			<-ticker
+			select {
+			case <-ticker:
+			case <-w.syncNow:
+				level.Info(w.logger).Log("msg", "explicit DEP sync requested")
```

Comment:
why is there no action here? is this a stub?

Reply:
No action? The point of the select (and the unconditional wait before it) is merely to wait for the ticker to expire or the SyncNow signal to fire to return early to do another sync again.